pygotham/frontend/__init__.py

```diff
@@ -61,6 +64,51 @@ def create_app(settings_override=None):
     def current_event():
         return {'current_event': get_current_event()}
 
+    @app.context_processor
+    def generate_navbar():
+        """Autodiscover links that should populate the site's navbar."""
+        # navbar_links is a dict of the form
+        # {section_name: {link_name: link_value, ...}, ...}
+        navbar_links = defaultdict(dict)
+
+        # First, autogenerate a list based on available routes
+        # Available routes are any that have a GET method and have all
+        # arguments (if any) provided default values
+        for rule in app.url_map.iter_rules():
+            section_name = rule.endpoint.split('.')[0]
+            get_allowed = 'get' in (method.lower() for method in rule.methods)
+            required_args = set(rule.arguments)
+            provided_args = set(getattr(rule.defaults, 'keys', lambda: [])())
```

Comment:
Total bikeshed, but this line takes a minute to understand. What about an approach like

    try:
        provided_args = set(rule.defaults.keys())
    except AttributeError:
        provided_args = set()